pydantic/types.py

```diff
@@ -2393,9 +2393,12 @@ def __get_pydantic_json_schema__(
         return field_schema
 
     def __get_pydantic_core_schema__(self, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        original_schema = handler(source)
+        # bytes schema has no keyword argument type.
+        original_schema.pop("type", None)
         return core_schema.with_info_after_validator_function(
             function=self.decode,
-            schema=core_schema.bytes_schema(),
+            schema=core_schema.bytes_schema(**original_schema),
```

Comment:
Can't we just do:

```suggestion
        return core_schema.with_info_after_validator_function(
            function=self.decode,
            schema=handler(source),
        )
```

and maybe additionally raise if `handler(source)` is not a bytes schema?

Reply:
update it